dNet/WorldPackets.h

```diff
@@ -2,40 +2,75 @@
-	uint16_t port = 80;
-	bool openWeb = false;
-	bool supportsSum = false;
-	bool supportsDetail = false;
-	bool supportsWho = false;
-	bool supportsObjects = false;
-	void Serialize(RakNet::BitStream &bitstream) const;
+class Entity;
+enum class eLanguageCodeID : int32_t {
+	EN_US = 0,
+	PL_US = 1,
+	DE_DE = 2,
+	EN_GB = 3
 };
 
 namespace WorldPackets {
-	void SendLoadStaticZone(const SystemAddress& sysAddr, float x, float y, float z, uint32_t checksum, LWOZONEID zone);
-	void SendCharacterCreationResponse(const SystemAddress& sysAddr, eCharacterCreationResponse response);
-	void SendCharacterRenameResponse(const SystemAddress& sysAddr, eRenameResponse response);
-	void SendCharacterDeleteResponse(const SystemAddress& sysAddr, bool response);
-	void SendTransferToWorld(const SystemAddress& sysAddr, const std::string& serverIP, uint32_t serverPort, bool mythranShift);
-	void SendServerState(const SystemAddress& sysAddr);
-	void SendCreateCharacter(const SystemAddress& sysAddr, int64_t reputation, LWOOBJID player, const std::string& xmlData, const std::u16string& username, eGameMasterLevel gm);
-	void SendChatModerationResponse(const SystemAddress& sysAddr, bool requestAccepted, uint32_t requestID, const std::string& receiver, std::set<std::pair<uint8_t, uint8_t>> unacceptedItems);
-	void SendGMLevelChange(const SystemAddress& sysAddr, bool success, eGameMasterLevel highestLevel, eGameMasterLevel prevLevel, eGameMasterLevel newLevel);
-	void SendHTTPMonitorInfo(const SystemAddress& sysAddr, const HTTPMonitorInfo& info);
-	void SendDebugOuput(const SystemAddress& sysAddr, const std::string& data);
+
+	struct UIHelpTop5: public LUBitStream {
+		eLanguageCodeID languageCode = eLanguageCodeID::EN_US;
+
+		UIHelpTop5() : LUBitStream(eConnectionType::WORLD, MessageType::World::UI_HELP_TOP_5) {};
+		bool Deserialize(RakNet::BitStream& bitStream) override;
+		void Handle() override;
+	};
+
+	struct GeneralChatMessage : public LUBitStream {
+		uint8_t chatChannel = 0;
+		std::u16string message;
+
+		GeneralChatMessage() : LUBitStream(eConnectionType::WORLD, MessageType::World::GENERAL_CHAT_MESSAGE) {};
+		bool Deserialize(RakNet::BitStream& bitStream) override;
+		void Handle() override;
+	};
+
+	struct PositionUpdate : public LUBitStream {
+		NiPoint3 position = NiPoint3Constant::ZERO;
+		NiQuaternion rotation = NiQuaternionConstant::IDENTITY;
+		bool onGround = false;
+		bool onRail = false;
+		NiPoint3 velocity = NiPoint3Constant::ZERO;
+		NiPoint3 angularVelocity = NiPoint3Constant::ZERO;
+		struct LocalSpaceInfo {
+			LWOOBJID objectId = LWOOBJID_EMPTY;
+			NiPoint3 position = NiPoint3Constant::ZERO;
+			NiPoint3 linearVelocity = NiPoint3Constant::ZERO;
+		};
+		LocalSpaceInfo localSpaceInfo;
+		struct RemoteInputInfo {
+			bool operator==(const RemoteInputInfo& other) {
+				return m_RemoteInputX == other.m_RemoteInputX && m_RemoteInputY == other.m_RemoteInputY && m_IsPowersliding == other.m_IsPowersliding && m_IsModified == other.m_IsModified;
+			}
```

Comment:
```suggestion
			bool operator==(const RemoteInputInfo&) = default;
```

Reply:
Does this just work then? or does it need to be defined somewhere else?

Reply:
this works as is - =default means "use a compiler generated == comparison" which will just compare all members for equality with other

Reply:
![image](https://github.com/user-attachments/assets/e6686c92-aa85-4c4b-a7b9-c34e86250686)
It does not like that

Reply:
(dicord) It's const so that' doesn't work